java-tools/OsmAndServer/src/main/java/net/osmand/server/tileManager/GeotiffTile.java

```diff
@@ -27,8 +29,22 @@ public GeotiffTile(TileServerConfig cfg, GeotiffTileController.TileType tileType
 		touch();
 	}
 
+	public void setRuntimeImage(BufferedImage runtimeImage) {
+		this.runtimeImage = runtimeImage;
+	}
+
 	public void touch() {
 		lastAccess = System.currentTimeMillis();
+		File cacheFile = getCacheFile(".png");
+		if (cacheFile != null && cacheFile.exists()) {
+			long lastModifiedTime = cacheFile.lastModified();
+			if (lastAccess > lastModifiedTime + THREE_HOURS_IN_MILLIS) {
+				boolean success = cacheFile.setLastModified(lastAccess);
+				if (!success) {
+					lastAccess = lastModifiedTime;
+				}
+			}
+		}
```

Comment:
When the method called "touch" it's OK to setLastModified() every time without additional 3hrs-check.